src/utils/stringifyRules.js

```diff
@@ -43,25 +43,63 @@ const parseRulesPlugin = _insertRulePlugin(rule => {
   parsingRules.push(rule);
 });
 
-stylis.use([parseRulesPlugin, returnRulesPlugin]);
+let _componentId: string;
+let _selector: string;
+let _selectorRegexp: RegExp;
+
+const selfReferenceReplacer = (match, offset, string) => {
+  if (
+    // the first self-ref is always untouched
+    offset > 0 &&
+    // there should be at least two self-refs to do a replacement (.b > .b)
+    string.slice(0, offset).indexOf(_selector) !== -1 &&
+    // no consecutive self refs (.b.b); that is a precedence boost and treated differently
+    string.slice(offset - _selector.length, offset) !== _selector
+  ) {
+    return `.${_componentId}`;
+  }
+
+  return match;
+};
+
+/**
+ * When writing a style like
+ *
+ * & + & {
+ *   color: red;
+ * }
+ *
+ * The second ampersand should be a reference to the static component class. stylis
+ * has no knowledge of static class so we have to intelligently replace the base selector.
+ */
+const selfReferenceReplacementPlugin = (context, _, selectors) => {
+  if (context === 2 && selectors[0].lastIndexOf(_selector) > 0) {
```

Comment:
We've had to revert to 4.0.0 in buildkite/site#234 because this line seems to have broken vendor prefix of webfonts? The arguments when `selectors[0]` was undefined is:

```
{ context: 2,
  selectors: [],
  _:
   'font-family:BK-Circular;src:url(/_next/static/assets/assets/fonts/lineto-circular-book-0415b07a.woff2) format("woff2");font-weight:normal;font-style:normal;font-display:swap;' }
```

I wonder if the vendor prefixing `context` of 2 is being confused with the plugin `context` of 2?

Reply:
If you could provide a code sample I could use in a unit test, that'd be great. In the meantime, I made this PR that should work around the specific error you mentioned: https://github.com/styled-components/styled-components/pull/2114

Reply:
Thanks @probablyup 🙏🏼 That fixed it.

I've tried for an hour to create the smallest possible test case (both browser rendered and server rendered), but failed to recreate it I'm sorry. Crummy bug report I know 😔

Here's the commands to recreate it exactly though, if you needed:

```bash
git clone https://github.com/buildkite/site.git
cd site && git checkout da15e8cd
yarn && yarn start
```